### ISSUE TITLE
Emulator reopens other instances's windows

### ISSUE THREAD

**Issue by Colpocorto** [openMSX/openMSX#1777]
I'm not sure if this is a bug or it is by design, indeed (in any case I find it very annoying).

When you work with multiple instances, the same already opened windows will open when you open a new emulator.
This behaviour floods the screen with windows and even the main emulator window is covered by them when you boot the machine.

To my understanding, debugging or other windows opened by an instance should not open automatically when you open a NEW instance that is not necessary related to the one already opened.

**Comment by MBilderbeek** [comment]
The window layout is always saved and a new instance simply loads the same (just saved) layout as well.

What do you propose as behaviour instead?  How should openMSX know whether or not to load the data from the imgui.ini file?

**Comment by Colpocorto** [comment]
I suggest adding a flag that just loads (or not) the whole bunch of windows currently opened by a previous instance. If would be a checkbox in the menu, something like

"Settings > GUI > ☑ Reopen previous windows"

I think the cost is cheap and would make everyone happy.

**Comment by MBilderbeek** [comment]
But that flag would then only have effect the *next* time you open the application, right?

**Comment by m9710797** [comment]
I understand the request. But I think it's more complicated than you think.
I think we should first clean up what information gets stored where (and maybe create some additional/separate file).
At the moment everything GUI related is stored in `imgui.ini`. That includes the 'Dear ImGui' stuff itself e.g. last sizes and positions of windows (whether they currently open or not), how windows are docked, ...
We extended that file with information on: which windows are open, font type and size, recently used files, debug settings (like prefered grid color etc).
Next to the `imgui.ini` file we have `settings.xml`. This contains all settings that existed prior to openMSX-20. So the separation is mostly historical rather than functional.
I think we should first better think about what are openMSX settings, Dear imgui settings, (other settings?).  What should be part of the "session". And only then can we think about how to deal with multiple sessions (e.g. whether to auto-restore the session or not).

**Comment by MBilderbeek** [comment]
Yes, I was thinking the same: we first need to rethink these settings. Ideas most welcome.

**Comment by sdsnatcher** [comment]
> "Settings > GUI > ☑ Reopen previous windows"
> 
> I think the cost is cheap and would make everyone happy.

I agree that this is a simple solution that would cover most use cases.

My use case (creating patches for old games) requires a lot of reverse engineering, guessing, experimentation, retries, rinse-and-repeat. This means I need the debugger windows/breakpoints/watchpoints/etc to be restored every time. It’s easier to just close/delete what I don’t need on another session than redo everything each time.

**Comment by MBilderbeek** [comment]
See also #1698 ...
